internal/engine/engine.go

```diff
@@ -587,22 +587,33 @@ func (engine *Engine) getPartialRuleTable(ctx context.Context, resource, policyV
 	// Therefore, we need to retrieve the compilation unit for each scope, remove all bar the first policy,
 	// and pass all individually to the LoadPolicies method.
 	toLoad := []*runtimev1.RunnablePolicySet{}
+	// we force lenientScopeSearch when retrieving resource policy sets as lenient scope search is enforced
+	// in the evaluator function. Therefore, to prevent duplicate rows in the rule table, we check the returned
+	// policy scope before adding to `toLoad`
+	addedResourceScopes := make(map[string]struct{})
 	addScopedPolicyRules := func(scope string) error {
 		rps, err := engine.getResourcePolicySet(ctx, resource, policyVer, scope, true)
 		if err != nil {
 			return err
 		}
 
 		if rps != nil {
-			toLoad = append(toLoad, rps)
+			// check the first policy scope
+			for _, p := range rps.GetResourcePolicy().GetPolicies() {
```

Comment:
Why not just take the first policy using `rps.GetResourcePolicy().GetPolicies()[0]`?

Does this also mean that it's no longer necessary for us to include all the ancestors of the policy in the policy set? If so, we should think about deprecating that format and making it simpler (in a future release).

Reply:
For the "just in case" implicit nil checking the `Get*` methods provide, really. But yeah, given the known type I could just do this!